src/plugins/kibana/public/dashboard/components/panel/panel.html

```diff
@@ -1,14 +1,31 @@
 <div class="panel panel-default" ng-switch on="panel.type" ng-if="savedObj || error">
   <div class="panel-heading">
-    <span class="panel-title">
+    <div class="panel-title">
       <i
         class="fa"
         ng-class="savedObj.vis.type.icon"
         aria-label="{{savedObj.vis.type.title}} Icon"
         title="{{savedObj.vis.type.title}}">
       </i>
-      {{savedObj.title}}
-    </span>
+      <a class="action panel-title-edit" ng-click="setTitle()">
+	      <span ng-if="panel.customTitle">{{panel.customTitle}}</span>
+	      <span ng-hide="panel.customTitle">{{savedObj.title}}</span>
+      </a>
+      <input size="10" ng-if="showFormTitle" text="title" ng-model="panel.customTitle" ng-show="chrome.getVisible() && editUrl">
+
+     <span ng-if="showFormTitle" class="title-actions" ng-show="chrome.getVisible() && editUrl">
```

Comment:
- Again, editUrl is not required here.
- This would look better as part of an `input-group`